sympy/simplify/fu.py

```diff
@@ -2103,6 +2113,9 @@ def hyper_as_trig(rv):
         t, f = hyper_as_trig(expr)
         expr == f(t)
 
+    Be careful in simplifying returned expression as some trigonometric
```

Comment:
This makes me nervous -- can you give an example?

Reply:
I see #11685, but didn't you fix TR10i so now this warning wouldn't apply?

Reply:
@smichr As TR10i fixes this by undocumented private _hyper flag, it is not directly applicable. Also, I think it is possible that user or we may implement new trigonometric simplification or simplify the expression themselves.

Reply:
Aren't all manipulations of the transformed hyperbolic functions valid? I don't know that much about it so forgive the ignorance.

Reply:
@smichr When transforming hyperbolic functions to trigonometric functions and vice versa, `I` is multiplied to the arguments. The `hyper_as_trig()` does this by multiplying dummy variable when transforming hyperbolic functions to trigonometric functions, and substituting dummy variable with -1 when transforming back, assuming argument is multiplied by the dummy variable. However, this fails when constant is introduced to the arguments, which should be multiplied by I when transforming back.

Reply:
Ahh...that makes sense. Can't we fix this by making sure there is no numerical coefficient on the result when we convert back to hyperbolics? e.g. if we had j*f(x) and end up with c + j*g(x) after simplification, then can't we replace c with c*j? But can you give a specific example of such a c being introduced?

Reply:
@smichr `I*sqrt(3)*cosh(x)/2 + sinh(x)/2` -_osborne-> `I*sqrt(3)*cos(d*x)/2 + I*sin(d*x)/2` -TR10i-> `I*sin(d*x + pi/3)` -_osbornei-> `sinh(x + pi/3)` where `I*sqrt(3)*cosh(x)/2 + sinh(x)/2 == sinh(x + pi/3*I)`.

Reply:
@smichr Maybe like below?:
```diff
diff --git a/sympy/simplify/fu.py b/sympy/simplify/fu.py
index d07bffbfe..170397520 100644
--- a/sympy/simplify/fu.py
+++ b/sympy/simplify/fu.py
@@ -2084,7 +2084,8 @@ def _osbornei(e, d):
     def f(rv):
         if not isinstance(rv, TrigonometricFunction):
             return rv
-        a = rv.args[0].xreplace({d: S.One})
+        const, a = rv.args[0].as_independent(d)
+        a = a.xreplace({d: S.One}) + const*I
         if rv.func is sin:
             return sinh(a)/I
```

Reply:
Yes -- but use flag `as_Add=True` with `as_independent` to ensure that you are getting an additive constant.

Also, I am having a hard time following the example that you gave. Could you show me an explicit interactive session where you end up with that additive constant?